app/V1Module/presenters/PipelinesPresenter.php

```diff
@@ -148,15 +148,22 @@ public function actionCreatePipeline() {
       $exercise = $this->exercises->findOrThrow($this->getRequest()->getPost("exerciseId"));
     }
 
-    if (!$this->pipelineAcl->canCreate() ||
-        ($exercise && !$this->exerciseAcl->canCreatePipeline($exercise))) {
+    if (!$this->pipelineAcl->canCreate()) {
       throw new ForbiddenRequestException("You are not allowed to create pipeline.");
     }
+    if ($exercise && !$this->exerciseAcl->canAttachPipeline($exercise)) {
+      throw new ForbiddenRequestException("You are not allowed to attach newly created pipeline to given exercise.");
+    }
 
     // create pipeline entity, persist it and return it
-    $pipeline = Pipeline::create($this->getCurrentUser(), $exercise);
+    $pipeline = Pipeline::create($this->getCurrentUser());
     $pipeline->setName("Pipeline by {$this->getCurrentUser()->getName()}");
     $this->pipelines->persist($pipeline);
+    if ($exercise) {
+      $exercise->addPipeline($pipeline);
+      $this->exercises->persist($exercise);
+      $this->pipelines->refresh($pipeline); // this is needed so pipeline contains the newly added exercise
```

Comment:
You maybe should keep the original signature of `create` method on `Pipeline` and handing over only single `Exercise` and construct the `Pipeline` with proper `Collection` and not null. Therefore this whole `if` would not be necessary. The same approach was used in `Exercise` <-> `Group` relation if I am not mistaken.